### PR TITLE
Jquery testing

### PR DESCRIPTION
**Pull request by BryonLewis**
**TLDR**: Jquery bug in how `.trigger` operates causes an issue on the checkboxes that have their change event is bound to re-render (destroy/recreate) of the element.  Changed the test to utilize DOM native click instead of jQuery click.  Further in depth explanation below.

Changes the JobsList test to fix an issue with Jquery and the way the JobsList renders.
The JobsList has a connection between all of the checkboxes in the list and the parent check/uncheck all button and actions.  Whenever a button is checked it will toggle the state of parent checkbox.  To do this it re-renders the entire list each time a checkbox is clicked and calculates if none, any, or all boxes are checked.
GirderJobsList

An update to jQuery caused the `.trigger` and `.click` methods to attempt to access an non-existent data member.
Specific Change that caused the issue: [Code Link](https://github.com/jquery/jquery/commit/24d71ac70406f522fc1b09bf7c4025251ec3aee6?diff=unified#diff-031bb62d959e7e4949d1847c82507f33L579-R583) in PR [4354](https://github.com/jquery/jquery/pull/4354)

There is a current in process PR to fix the issue which may work it's way through:  [Fixing in Process PR #4725](https://github.com/jquery/jquery/pull/4725)
After the fix we could revert the change to the test to utilize `.trigger` again.

Other option is a future refactor of how the checkboxes function in the JobList.

### CODE REVIEW

**Comment by brianhelba** [comment]
@BryonLewis Could you rebase this down to 1-2 commits (perhaps one upgrading and one fixing the issue)? There's no need to include intermediate development history.

Otherwise, this LGTM! Thanks again for figuring this out.
